ansible/roles/mongodb-backup/tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+- name: Create the directory
+  file: path={{ mongo_backup_dir }} state=directory recurse=yes
+  
+- set_fact:
+    mongo_backup_file_name: "mongo-backup-{{ ansible_hostname }}-{{ lookup('pipe', 'date +%Y-%m-%d') }}"
+ 
+- set_fact:
+    mongo_backup_file_path: "{{ mongo_backup_dir }}/{{ mongo_backup_file_name }}"
+
+- name: take dump
+  shell: "mongodump --out={{ mongo_backup_file_path }}"
+    
+- name: Compress the backup file
+  shell: "zip -r {{ mongo_backup_file_path }}.zip {{ mongo_backup_file_path }}"
+
+- name: upload to azure
+  include_role:
+    name: artifacts-upload-azure
+  vars:
+    artifact: "{{ mongo_backup_file_name }}.zip"
+    artifact_path: "{{ mongo_backup_file_path }}.zip"
+    artifacts_container: "{{ mongo_backup_azure_container_name }}"
```

Comment:
Does this role create the container if it doesn't exist?

Reply:
yes @keshavprasadms . It will create container in blob.